### PR TITLE
[RFR] Creating basic structure of application

### PR DESCRIPTION
**Pull request by JulienMattiussi**
Defining pages of the game and basic navigation between them

image

### CODE REVIEW

**Comment by djhi** [comment]
What's left for RFR ? Can you add a checklist in your PR description ?

**Review by djhi — CHANGES_REQUESTED** [review]
Can you add eslint and prettier ? Ping me if you need help to configure them correctly

**Review by djhi — CHANGES_REQUESTED** [review]
I'm not fan of the types oriented files organization as it usually makes it a lot harder to reason about a feature. You should create a folder for each route and keep all the components specific to each route in its folder. If you have some components used in more than one route, introduce a `lib` or `components` folder at the project src

File: routes/GameListScreen.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          onPress={this.backHome}
+          title="Back to home"></Button>
+      </View>
+    );
+  }
+
+  backHome = async mode => {
+    try {
+      const { navigation } = this.props;
+      navigation.navigate('Home', {
```
Comment (djhi) [inline routes/GameListScreen.js]: Why the empty object parameter ?

File: routes/GameScreen.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          onPress={this.backHome}
+          title="Back to home"></Button>
+      </View>
+    );
+  }
+
+  backHome = async mode => {
+    try {
+      const { navigation } = this.props;
+      navigation.navigate('Home', {
```
Comment (djhi) [inline routes/GameScreen.js]: Why the empty object parameter ?

File: routes/HomeScreen.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+          onPress={this.showGameList}
+          title="Show existing games"></Button>
+      </View>
+    );
+  }
+
+  openGame = async mode => {
+    try {
+      const { navigation } = this.props;
+      navigation.navigate('Game', {
```
Comment (djhi) [inline routes/HomeScreen.js]: Why the empty object parameter ?

File: routes/HomeScreen.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        ToastAndroid.LONG,
+        ToastAndroid.BOTTOM,
+      );
+    }
+  };
+
+  showGameList = async mode => {
+    try {
+      const { navigation } = this.props;
+      navigation.navigate('GameList', {
```
Comment (djhi) [inline routes/HomeScreen.js]: Why the empty object parameter ?

**Review by djhi — CHANGES_REQUESTED** [review]

**Review by djhi — COMMENTED** [review]
Don't forget to run eslint on all your files

```sh
./node_modules/.bin/eslint --fix ./src
```

**Review by djhi — CHANGES_REQUESTED** [review]
One last thing, we need a travis.yml file

**Review by djhi — APPROVED** [review]

**Comment by JulienMattiussi** [comment]
Slack est down, d'ou le mail
Tu a un peu de temps pour un call ?

https://appear.in/quarto-android



2018-06-27 11:39 GMT+02:00 Gildas Garcia <notifications@github.com>:

**Pull request merged**